src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -0,0 +1,140 @@
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+using Workleap.OpenApi.MSBuild.Exceptions;
+
+namespace Workleap.OpenApi.MSBuild;
+
+internal sealed class SpectralManager : ISpectralManager
+{
+    private const string SpectralVersion = "6.11.0";
+
+    private readonly IProcessWrapper _processWrapper;
+    private readonly ILoggerWrapper _loggerWrapper;
+    private readonly string _openApiToolsDirectoryPath;
+    private readonly string _toolDirectory;
+
+    public SpectralManager(IProcessWrapper processWrapper, ILoggerWrapper loggerWrapper, string openApiToolsDirectoryPath)
+    {
+        this._processWrapper = processWrapper;
+        this._loggerWrapper = loggerWrapper;
+        this._openApiToolsDirectoryPath = openApiToolsDirectoryPath;
+        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, $"spectral/{SpectralVersion}");
+    }
+
+    public async Task InstallSpectralAsync(CancellationToken cancellationToken)
+    {
+        this.CreateRequiredDirectories();
+        var executableFileName = this.GetSpectralFileName();
+
+        var url = $"https://github.com/stoplightio/spectral/releases/download/v{SpectralVersion}/{executableFileName}";
+
+        await this.DownloadFileAsync(url, $"{this._toolDirectory}/{executableFileName}", cancellationToken);
+    }
+
+    private void CreateRequiredDirectories()
+    {
+        Directory.CreateDirectory(Path.Combine(this._openApiToolsDirectoryPath, "spectral"));
+        Directory.CreateDirectory(this._toolDirectory);
+    }
+
+    private string GetSpectralFileName()
+    {
+        var osType = GetOperatingSystem();
+        var architecture = GetArchitecture();
+
+        if (osType == "linux")
+        {
+            var distro = File.Exists("/etc/os-release") ? File.ReadAllText("/etc/os-release") : string.Empty;
```

Comment:
Accessing the file system as it is right now isn't really unit testable but I'm fine for now as we don't have unit tests yet

Reply:
Will consider this and the unit test as a technical debt.